foxtrot-core/src/main/java/com/flipkart/foxtrot/core/table/impl/FoxtrotTableManager.java

```diff
@@ -0,0 +1,104 @@
+package com.flipkart.foxtrot.core.table.impl;
+
+import com.flipkart.foxtrot.common.Table;
+import com.flipkart.foxtrot.core.datastore.DataStore;
+import com.flipkart.foxtrot.core.datastore.DataStoreException;
+import com.flipkart.foxtrot.core.querystore.QueryStore;
+import com.flipkart.foxtrot.core.table.TableManager;
+import com.flipkart.foxtrot.core.table.TableManagerException;
+import com.flipkart.foxtrot.core.table.TableMetadataManager;
+
+import java.util.List;
+
+/**
+ * Created by rishabh.goyal on 05/12/15.
+ */
+public class FoxtrotTableManager implements TableManager {
+
+    private final TableMetadataManager metadataManager;
+    private final QueryStore queryStore;
+    private final DataStore dataStore;
+
+    public FoxtrotTableManager(TableMetadataManager metadataManager, QueryStore queryStore, DataStore dataStore) {
+        this.metadataManager = metadataManager;
+        this.queryStore = queryStore;
+        this.dataStore = dataStore;
+    }
+
+
+    @Override
+    public void save(Table table) throws TableManagerException {
+        validateTableParams(table);
+        try {
+            if (metadataManager.exists(table.getName())) {
+                throw new TableManagerException(TableManagerException.ErrorCode.BAD_REQUEST, "table already exists");
+            }
+            queryStore.init(table.getName());
+            dataStore.init(table);
+            metadataManager.save(table);
+        } catch (DataStoreException e) {
+            switch (e.getErrorCode()) {
+                case TABLE_NOT_FOUND:
+                    throw new TableManagerException(TableManagerException.ErrorCode.BAD_REQUEST, e);
+                case STORE_CONNECTION:
+                    throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+                default:
+                    throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+            }
+        } catch (TableManagerException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+        }
+    }
+
+    @Override
+    public Table get(String name) throws TableManagerException {
+        try {
+            Table table = metadataManager.get(name);
+            if (table == null) {
+                throw new TableManagerException(TableManagerException.ErrorCode.TABLE_NOT_FOUND, "table_does_not_exists");
```

Comment:
exist .. and why the underscores? Also, add the tablename for which this error is coming.